### DISCUSSION TITLE
Use of "Restrict who can push to matching branches" in the merge queue world #62541

### DISCUSSION THREAD

**Discussion by nikosharp** [community/community#62703] [Pull Requests] [answered]
### Select Topic Area

Product Feedback

### Body

I have some confusion around "Restrict who can push to matching branches" with merge queues.

The wording of "Organization administrators, repository administrators, the merge queue, and users with the Maintain role." implies to me that the merge queue has access to push to the branch, so that individual users do not need it - this is how we were using it with our previous merge queue vendor, where their app was given push access, but engineers were not.

However, I'm seeing that we do need to give push access to everyone to allow them to queue up their PRs into the merge queue. Without it, they see

Merging is blocked
Base branch restricts merging to restricted users

Is this expected behavior? I'd like to keep the restriction, though I do understand that since requiring the merge queue itself is a form of branch protection, as long as they aren't given force push, they aren't being given any dangerous permissions.

If this is expected, can "Restrict who can push to matching branches" be reworded to include even adding to the merge queue? And what does the inclusion of "the merge queue" in the blurb mean?

**Comment by mujeeb00** [comment, marked as answer]
The behavior is expected. To use the merge queue, everyone needs push access, but you can maintain the restriction by not granting force push permissions. The wording in the documentation could be clearer regarding the need for push access for the merge queue to function.

**Reply by Sahnesiguzel33** [reply]
malesef
